node_manager.go

```diff
@@ -0,0 +1,991 @@
+/*
+ * File: node_manager.go
+ *
+ * Description: functions to handle nodes and node creation
+ */
+
+package main
+
+import (
+	"encoding/json"
+	"path"
+	"strconv"
+	"strings"
+	"time"
+
+	clientv3 "github.com/coreos/etcd/clientv3"
+	"github.com/coreos/etcd/mvcc/mvccpb"
+	"golang.org/x/net/context"
+)
+
+//
+// Node definition
+//
+type Node struct {
+	HostName string
+	HostID   string
+}
+
+//! Update the TTL for a given reserved entry
+/*
+ * @param    LeaseID    id of a given etcd key entry
+ *
+ * @return   none
+ */
+func (inst *EtcdInstance) updateTTL(leaseID clientv3.LeaseID) {
+
+	// keep the entry alive...
+	_, err := inst.internal.KeepAliveOnce(context.TODO(), leaseID)
+
+	// if an error occurs, dump it to stdout
+	if err != nil {
+		stdlog(err.Error())
+		stdlog("updateTTL() --> could not update TTL")
+	}
+}
+
+//! Function to keep etcd key values alive
+/*
+ * @param    LeaseGrantResponse    lease response
+ *
+ * @return   none
+ */
+func (inst *EtcdInstance) keepKeyAlive(lease *clientv3.LeaseGrantResponse) {
+
+	// set a duration time based on the time-to-live
+	sleep_duration := time.Duration(lease.TTL / 2)
+
+	// infinite loop
+	for {
+
+		// update the time-to-live
+		inst.updateTTL(lease.ID)
+
+		// then go back to sleep
+		time.Sleep(time.Second * sleep_duration)
```

Comment:
Admittedly not the best logic, I have since corrected it.

https://github.com/rbisewski/lclusterd/commit/ea367cd2859f4299d606bf318f57ef52c806c84d